wgpu-types/src/lib.rs

```diff
@@ -4584,6 +4584,10 @@ pub struct SurfaceCapabilities {
     ///
     /// Will return at least one element, CompositeAlphaMode::Opaque or CompositeAlphaMode::Inherit.
     pub alpha_modes: Vec<CompositeAlphaMode>,
+    /// List of supported texture usages for the surface to use with the given adapter.
+    ///
+    /// What happens if the surface is incompatible with the adapter?
```

Comment:
It should panic or give an error in the error handler if one is installed. We should add that to the list of error cases on `wgpu::Adapter::configure`

Reply:
comment here should also mention that render target usage is guaranteed (just like the corresponding HAL field does :))

Reply:
Does this mean that the default implementation should be RENDER_ATTACHMENT rather than empty()?

I assumed that the error reporting is already handled elsewhere. The comments mention what the fields are in the case of incompatibility though. I assume that that is the purpose of the Default implementation, since it matches the comments?

Reply:
> Does this mean that the default implementation should be RENDER_ATTACHMENT rather than empty()?

I think so, yes!